### PR TITLE
Add Continuous Integration

### PR DESCRIPTION
**Pull request by bzz**
Fixes first 2 items from #72:

 - check for link duplicates in README
 - check that links are accessible
 - fixes paper link duplication

CI is [expected to fail](https://travis-ci.org/bzz/awesome-machine-learning-on-source-code/builds/455212425), as there are some broken links in README right now.

If we agree on approach of dealing with broken links (remove/replace/etc) I can fix those here, but would prefer to deal with them in a separate PR.

----

[dkhamsing/awesome_bot](https://github.com/dkhamsing/awesome_bot) was chosen after a small research

- https://github.com/raviqqe/liche
  - ✅  fast, written in Go
  - 🚫 fails with
  > Error when reading response headers: small read buffer. Increase ReadBufferSize. Buffer size=4096

  needs refactoring to set [ReadBufferSize](https://github.com/valyala/fasthttp/blob/master/client.go#L195)

 - https://github.com/tcort/markdown-link-check
   - 🚫 fails on redirects
   - 🐌 slow

### CODE REVIEW

**Review by m09 — APPROVED** [review]
Very nice. I think we should wait for the CI to pass before merging though, otherwise it will be confusing for external contributors, wdyt @bzz?

**Comment by m09** [comment]
Good for me now.

**Comment by m09** [comment]
Sorry I just re-read your initial PR description and you specify that you didn't want to fix the links in this one. I can remove my commits if you want, I didn't see it when I fixed the links. Still I think it might be good to have a non-failing CI on master.

**Comment by bzz** [comment]
Thank you for taking care @m09 - looks great to me!

Few more improvements:
 - added badge to README 
 - scheduled a daily cron job \w CI of master, so we could notice \wo a PR if something gets broken

should be good to merge, as soon as @EgorBu or @vmarkovtsev have time to ✅.

BTW right now all `https://dl.acm.org` links are not accessible, can not open any of it in browser and CI says:

```
  1. [L141] 522 https://dl.acm.org/citation.cfm?id=3180167  
  2. [L148] 522 https://dl.acm.org/citation.cfm?id=2892661  
  3. [L183] 522 https://dl.acm.org/citation.cfm?id=3243130  
  4. [L213] 522 https://dl.acm.org/citation.cfm?id=3243132  
  5. [L230] 522 https://dl.acm.org/citation.cfm?id=3240471  
  6. [L238] 522 https://dl.acm.org/citation.cfm?id=3243131  
```

May be good candidates for manually transfer to http://archive.is/ as soon as available.

**Comment by m09** [comment]
Thanks for the cron, it'd be good indeed to fix stuff regularly to avoid having contributors fail CI for some reason independent from their contribution! Which leads me to ask: maybe we should keep the current job only for the cron and have a more elaborated one that fails only if the added links are problematic?

**Comment by bzz** [comment]
@m09 that is a nice improvement suggestion indeed! 

But as with any improvement - I would prefer merging the baseline first and then having a separate issue for refactoring/adding the improvement, addressing it when we stumble up on it as an actual problem instead of keeping this open forever and keep adding hypothetical improvements. 

WDYT? 

\cc @EgorBu @vmarkovtsev for review

File: README.md
Comment (EgorBu) [inline README.md:61]: this line could be left unmodified - but LGTM for me in both cases

File: README.md
Comment (EgorBu) [inline README.md:171]: I prefer link to university site over private github page

File: README.md
Comment (EgorBu) [inline README.md:236]: `ASE'16` looks better than `ASE%2716`

File: README.md
Comment (EgorBu) [inline README.md:349]: Now I see why you need `id="card2code"`

**Review by EgorBu — COMMENTED** [review]
Great job, it will help us a lot! Thank you! Couple of moments to discuss and it could be merged

File: README.md
Comment (bzz) [reply README.md:171]: @m09 this is exactly the reason I was proposing to have fixes and CI automation in separate PR

File: README.md
Comment (m09) [reply README.md:61]: As you can see the URL returns a `301` without the slash:
```
curl -i https://www.microsoft.com/en-us/research/event/artificial-intelligence-and-machine-learning-in-cambridge-2017
HTTP/1.1 301 Moved Permanently
Content-Type: text/html; charset=UTF-8
Location: https://www.microsoft.com/en-us/research/event/artificial-intelligence-and-machine-learning-in-cambridge-2017/
Strict-Transport-Security: max-age=31536000; includeSubDomains
X-Ten-Cache: MISS
Access-Control-Allow-Origin: https://microsoft.com, https://www.microsoft.com
X-Frame-Options: SAMEORIGIN
Content-Length: 0
X-EdgeConnect-Origin-MEX-Latency: 475
Expires: Tue, 20 Nov 2018 10:36:27 GMT
Cache-Control: max-age=0, no-cache, no-store
Pragma: no-cache
Date: Tue, 20 Nov 2018 10:36:27 GMT
Connection: keep-alive
X-RTag: RT
```

File: README.md
Comment (m09) [reply README.md:171]: @bzz I agree but anyway, we'll still manage!

@EgorBu I agree but the link has been redirected and we should honor that:
```
curl -i http://sei.pku.edu.cn/~xiongyf04/papers/ISSTA18a.pdf     
HTTP/1.1 302 Moved Temporarily
Server: nginx/1.6.2
Date: Tue, 20 Nov 2018 10:52:40 GMT
Content-Type: text/html
Content-Length: 160
Connection: keep-alive
Location: https://xiongyingfei.github.io/papers/ISSTA18a.pdf
```

File: README.md
Comment (m09) [reply README.md:236]: I agree but it causes problems with the CI scrit and has the same meaning otherwise. If you see a better solution I'm all for it :)

File: .travis.yml
Comment (vmarkovtsev) [inline .travis.yml:8]: Wow cool @bzz! Ruby devs automated even this...

File: README.md
Comment (EgorBu) [reply README.md:61]: ok, let's use `/`.

File: README.md
Comment (EgorBu) [reply README.md:171]: good point!

File: README.md
Comment (EgorBu) [reply README.md:236]: Didn't know about CI problem - fine in this case

**Pull request merged**
